src/hooks/useValidateTransactionMemo.ts

```diff
@@ -71,43 +77,49 @@ export const useValidateTransactionMemo = (incomingXdr?: string | null) => {
         return true;
       }
     },
-    [network],
+    [networkDetails.networkUrl],
   );
 
   useEffect(() => {
-    if (!transactionMemo) {
-      setIsMemoMissing(shouldValidateMemo);
+    if (!xdr || !network || !shouldValidateMemo) {
+      return;
     }
-  }, [transactionMemo, shouldValidateMemo]);
+
+    const transaction = TransactionBuilder.fromXDR(xdr, network);
+    const memo =
+      "memo" in transaction && transaction.memo.value
+        ? String(transaction.memo.value).trim()
```

Comment:
I'd suggest not trimming the memo string value just to avoid changing it without the user consent